src/app/clusters/identify-server/IdentifyCluster.cpp

```diff
@@ -0,0 +1,236 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/clusters/identify-server/IdentifyCluster.h>
+
+#include <app/InteractionModelEngine.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <clusters/Identify/Commands.h>
+#include <clusters/Identify/Metadata.h>
+#include <tracing/macros.h>
+
+#include <lib/support/logging/CHIPLogging.h>
+
+namespace chip::app::Clusters {
+
+using namespace Identify::Attributes;
+
+using namespace chip::app::Clusters::Identify;
+
+namespace {
+DefaultTimerDelegate sDefaultTimerDelegate;
+
+constexpr DataModel::AcceptedCommandEntry kAcceptedCommands[] = {
+    Identify::Commands::Identify::kMetadataEntry,
+};
+
+constexpr DataModel::AcceptedCommandEntry kAcceptedCommandsWithTriggerEffect[] = {
+    Identify::Commands::Identify::kMetadataEntry,
+    Identify::Commands::TriggerEffect::kMetadataEntry,
+};
+
+} // namespace
+
+IdentifyCluster::IdentifyCluster(const Config & config) :
+    DefaultServerCluster({ config.endpointId, Identify::Id }), mIdentifyTime(0), mPreviousIdentifyTime(0),
+    mIdentifyType(config.identifyType), mOnIdentifyStart(config.onIdentifyStart), mOnIdentifyStop(config.onIdentifyStop),
+    mOnEffectIdentifier(config.onEffectIdentifier), mCurrentEffectIdentifier(config.effectIdentifier),
+    mEffectVariant(config.effectVariant),
+    mTimerDelegate(config.timerDelegate == nullptr ? &sDefaultTimerDelegate : config.timerDelegate)
+{}
+
+DataModel::ActionReturnStatus IdentifyCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                             AttributeValueEncoder & encoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case Attributes::IdentifyTime::Id:
+        return encoder.Encode(mIdentifyTime);
+    case Attributes::IdentifyType::Id:
+        return encoder.Encode(mIdentifyType);
+    case Attributes::ClusterRevision::Id:
+        return encoder.Encode(Identify::kRevision);
+    case Attributes::FeatureMap::Id:
+        return encoder.Encode<uint32_t>(0);
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
+    }
+}
+
+DataModel::ActionReturnStatus IdentifyCluster::WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                              AttributeValueDecoder & decoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case Attributes::IdentifyTime::Id: {
+        mPreviousIdentifyTime = mIdentifyTime;
```

Comment:
You need to return a "no write" result when identify time is unchanged.

https://github.com/project-chip/connectedhomeip/blob/master/src/app/data-model-provider/ActionReturnStatus.h#L51